includes/class-wc-geolocation.php

```diff
@@ -136,25 +136,50 @@ public static function get_external_ip_address() {
 	 * @return array
 	 */
 	public static function geolocate_ip( $ip_address = '', $fallback = false, $api_fallback = true ) {
+		// Filter to allow custom geolocation of the IP address.
+		$country_code = apply_filters( 'woocommerce_geolocate_ip', false, $ip_address, $fallback, $api_fallback );
+
+		if ( false !== $country_code ) {
+			return array(
+				'country'  => $country_code,
+				'state'    => '',
+				'city'     => '',
+				'postcode' => '',
+			);
+		}
+
 		if ( empty( $ip_address ) ) {
 			$ip_address = self::get_ip_address();
 		}
 
 		$country_code = self::get_country_code_from_headers();
 
-		if ( false === $country_code ) {
-			// Filter to allow custom geolocation of the IP address.
-			$country_code = apply_filters( 'woocommerce_geolocate_ip', $country_code, $ip_address, $fallback, $api_fallback );
-		}
+		/**
+		 * Get geolocation filter.
+		 *
+		 * @since 3.9.0
+		 * @param array  $geolocation Geolocation data, including country, state, city, and postcode.
+		 * @param string $ip_address  IP Address.
+		 */
+		$geolocation  = apply_filters(
+			'woocommerce_get_geolocation',
```

Comment:
What about an action of `woocommerce_do_geolocation` or something, given that this is actually doing the geolocation work?